docs/genai/reference/config.md

```diff
@@ -49,128 +46,516 @@ A configuration file called genai_config.json is generated automatically if the
+  Additional config entries.
+
+- **provider_options**: *(array of ProviderOptions)*  
+  List of provider options.
+
+- **providers**: *(array of string)*  
+  List of providers to use at runtime.
+
+- **graph_optimization_level**: *(string, optional)*  
+  Graph optimization level.
+
+---
+
+#### ProviderOptions
+
+- **name**: *(string)*  
+  Name of the provider. One of:
+  - cuda
+  - DML
+  - NvTensorRtRtx
+  - OpenVINO
+  - QNN
+  - rocm
+  - WebGPU
+  - VitisAI
+
+  If this option is not given, the provider defaults to CPU.
+
+
+- **options**: *(array of [string, string] pairs)*  
+  Provider-specific options.
+
+---
+
+### Search
+
+Describes the generation/search parameters.
+
+- **do_sample**: *(bool)*  
+  Whether to use random sampling (top-k/top-p) or deterministic (greedy) search.
+
+- **min_length**: *(int)*  
+  Configured minimum length of the prompt + generated sequence, in the range of 0 to the context length of the model.
+
+- **max_length**: *(int)*  
+  Configured maximum length of the prompt + generated sequence, in the range of 0 to the context length of the model. If unspecified by the config or the user, this value is set to the context length of the model. `max_length` determines how much memory is allocated for the KV cache. 
+
+- **batch_size**: *(int)*  
+  Number of sequences to generate in parallel.
+
+- **num_beams**: *(int)*  
+  Number of beams for beam search. 1 means no beam search.
+
+- **num_return_sequences**: *(int)*  
+  Number of sequences to return.
+
+- **repetition_penalty**: *(float)*  
+  Penalty for repeating tokens. 1.0 means no penalty.
+
+- **top_k**: *(int)*  
+  Top-K sampling parameter.
+
+- **top_p**: *(float)*  
+  Top-P (nucleus) sampling parameter.
+
+- **temperature**: *(float)*  
+  Sampling temperature.
+
+- **early_stopping**: *(bool)*  
+  Whether to stop beam search early.
+
+- **no_repeat_ngram_size**: *(int)*  
+  Size of n-grams that should not be repeated.
+
+- **diversity_penalty**: *(float)*  
+  Not supported.
+
+- **length_penalty**: *(float)*  
+  Controls the length of the output. >1.0 encourages longer sequences, <1.0 encourages shorter.
+
+- **past_present_share_buffer**: *(bool)*  
+  If true, the past and present buffer are shared for efficiency.
+
+- **random_seed**: *(int)*  
+  Seed for the random number generator. -1 means use a random device.
+
+---
+
+## Notes
+
+- **session_options**:  
+  Supports advanced ONNX Runtime options such as threading, memory arena, quantization, profiling, and custom ops. See the C++ `SessionOptions` struct for all possible fields.
+
+- **inputs/outputs**:  
+  The names and patterns here must match the actual ONNX model graph.
+
+- **pipeline**:  
+  For advanced models, a pipeline of sub-models can be specified, each with its own ONNX file, session options, and input/output mappings.
+
+---
+
+## Search combinations
 
-2. Greedy search
+1. **Beam search**
+   - `num_beams > 1`
+   - `do_sample = false`
```

Comment:
not sure if we should mention the relationship between beam search and past present buffer sharing.

```suggestion
   - `do_sample = false`
   - `past_present_share_buffer = false`
```

Reply:
Because beam search needs to swap the order of the kv cache when num beams > 1 to keep the more relevant beams up front.
And swapping the order of the kv cache cannot be done when past present buffer sharing is enabled. So we implicitly will disable past present buffer sharing when num beams > 1.